index.js

```diff
@@ -138,6 +148,266 @@ app.intent('send.message - no', async conv => {
   conv.contexts.set('anything_else', 2);
 });
 
+/**
+ * join.conference
+ */
+app.intent('join.conference', async (conv, {target}) => {
+  const circuit = await getCircuit(conv);
+  if(!circuit){
+    return;
+  }
+
+  let matches = [];
+  let { inConferenceCall, conferenceAvailable, calls } = await getConferenceInfo(conv);
+
+  if (inConferenceCall) {
+    conv.ask("<speak>You're already in a conference call. Can I do anything else for you?</speak>");
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2); 
+    return;
+  } 
+
+  if(!conferenceAvailable){
+    conv.ask(`<speak>No conferences available to join. Can I do something else for you?</speak>`);
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  // If target isn't undefined
+  if(target){
+    calls.forEach(c => {
+      if(c.title.toLowerCase() === target.toLowerCase()){
+        matches.push(c);
+      }
+    });
+
+    if (!matches.length) {
+      let prompt = `<speak>I cannot find any conference call with name ${target}. Here are the names of your current ongoing conferences. 
+      Which would you like to join?</speak>`;
+      let titles = calls.map((call) => call.title.toLowerCase());
+      conv.ask(prompt, new Suggestions(titles));
+      conv.contexts.set('joinconference_gettarget', 5, {
+        calls: calls
+      });
+      return;
+    }
+
+    if (matches.length === 1) {
+      // One result found. Ask user for confirmation.
+      const confId = matches[0].callId;
+      conv.ask(`<speak>Ready to join the <break time="0.5s"/>${target}<break time="0.5s"/> conference call?</speak>`, new Suggestions('Yes', `No, don't join`));
+      conv.contexts.set('joinconference_send', 5, {
+        confId: confId
+      });
+      return;
+    }
+
+    // Multiple matches. Tell user that they can either re-enter a conference or just join the first match.
+    conv.ask(`More than one conference found with the name ${target}. Joining the first ${target} conference.`);
+    const confId = matches[0].callId;
+    conv.ask(`<speak>Ready to join the <break time="0.5s"/>${target}<break time="0.5s"/> conference call?</speak>`, new Suggestions('Yes', `No, don't join`));
+    conv.contexts.set('joinconference_send', 5, {
+      confId: confId
+    });
+    return;
+  }
+
+  let titles = [];
+  calls.forEach((call) => {
+    if(call.title === ""){
```

Comment:
if (!call.title) {

unless you purposely want to exclude checking for 'undefined', 'false', 0.